File: matiasmasso/Mat.NET/DT2/DTOCustomerCluster.vb
Public Class DTOCustomerCluster
    Inherits DTOGuidNom

    Property Holding As DTOHolding
    Property Obs As String

    Property Customers As List(Of DTOCustomer)

    Public Sub New()
        MyBase.New()
        _Customers = New List(Of DTOCustomer)
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
        _Customers = New List(Of DTOCustomer)
    End Sub

    Shared Shadows Function Factory(oHolding As DTOHolding) As DTOCustomerCluster
        Dim retval As New DTOCustomerCluster
        retval.Holding = oHolding
        Return retval
    End Function
End Class

File: matiasmasso/Mat.NET/DTO/DTOCustomerCluster.cs
using System;
using System.Collections.Generic;

namespace DTO
{
    public class DTOCustomerCluster : DTOGuidNom
    {
        public DTOHolding holding { get; set; }
        public string obs { get; set; }

        public List<DTOCustomer> Customers { get; set; }

        public DTOCustomerCluster() : base()
        {
            Customers = new List<DTOCustomer>();
        }

        public DTOCustomerCluster(Guid oGuid) : base(oGuid)
        {
            Customers = new List<DTOCustomer>();
        }

        public static DTOCustomerCluster Factory(DTOHolding oHolding)
        {
            DTOCustomerCluster retval = new DTOCustomerCluster();
            retval.holding = oHolding;
            return retval;
        }
    }
}
